### PR TITLE
picking terminal details

### PR DESCRIPTION
**Pull request by pvpurcarcosmin**
refs: https://github.com/metasfresh/metasfresh/issues/7101

### CODE REVIEW

File: backend/metasfresh-webui-api/src/main/java/de/metas/ui/web/picking/pickingslot/PickingSlotRepoQuery.java [hunk 1]
```diff
@@ -71,4 +75,18 @@ private PickingSlotRepoQuery(
 		this.onlyNotClosedOrNotRackSystem = onlyNotClosedOrNotRackSystem != null ? onlyNotClosedOrNotRackSystem : true;
 		this.pickingSlotBarcode = pickingSlotBarcode;
 	}
+
+	@Value
+	@Builder
+	public static class CreatePickingSlotRepoQueryReq
```
Comment (teosarca) [inline backend/metasfresh-webui-api/src/main/java/de/metas/ui/web/picking/pickingslot/PickingSlotRepoQuery.java:81]: why not moving this class to PickingSlotViewFactory and make it private?

File: backend/metasfresh-webui-api/src/main/java/de/metas/ui/web/picking/pickingslot/PickingSlotViewsIndexStorage.java [hunk 2]
```diff
@@ -175,14 +172,21 @@ public void closeById(@NonNull final ViewId pickingSlotViewId, @NonNull final Vi
...
-	public void invalidateView(ViewId pickingSlotViewId)
+	public void invalidateView(final ViewId pickingSlotViewId)
 	{
 		final PickingSlotView pickingSlotView = getOrCreatePickingSlotView(pickingSlotViewId, false/* create */);
 		if (pickingSlotView == null)
 		{
 			return;
 		}
 
+		final PackageableView packageableView = getPackageableViewByPickingSlotViewId(pickingSlotViewId);
```
Comment (teosarca) [inline backend/metasfresh-webui-api/src/main/java/de/metas/ui/web/picking/pickingslot/PickingSlotViewsIndexStorage.java:183]: pls document why we need to invalidate its parent view too (packageable view)?

File: backend/metasfresh-webui-api/src/main/java/de/metas/ui/web/picking/pickingslot/process/WEBUI_Picking_PickQtyToExistingHU.java [hunk 3]
```diff
@@ -134,20 +130,24 @@ public Object getParameterDefaultValue(@NonNull final IProcessDefaultParameter p
...
+				.map(IProductStorage::getProductId)
+				.collect(ImmutableSet.toImmutableSet());
+
+		final I_M_ShipmentSchedule selectedShipmentSchedule = getCurrentShipmentSchedule();
 
-		if (!orderIdOfThePickedHU.equals(orderIdOfTheCurrentSchedule))
+		if (productIds.size() != 1 || !productIds.contains(ProductId.ofRepoId(selectedShipmentSchedule.getM_Product_ID())))
 		{
-			throw new AdempiereException(msgBL.getTranslatableMsgText(MSG_WEBUI_PICKING_TO_THE_SAME_HU_FOR_MULTIPLE_ORDERS_NOT_ALLOWED));
+			throw new AdempiereException("VHUs cannot have multiple product storages!")
```
Comment (teosarca) [inline backend/metasfresh-webui-api/src/main/java/de/metas/ui/web/picking/pickingslot/process/WEBUI_Picking_PickQtyToExistingHU.java:148]: i think this error message applies only for productIds.size() != 1.
For the other part of the OR we need a different error message.

File: backend/metasfresh-webui-api/src/main/java/de/metas/ui/web/picking/pickingslot/PickingSlotRepoQuery.java (hunk 1 above)
Comment (pvpurcarcosmin) [reply backend/metasfresh-webui-api/src/main/java/de/metas/ui/web/picking/pickingslot/PickingSlotRepoQuery.java:81]: yeah, sure, why not

**Pull request merged**